tests/toranj/cli/test-035-context-id-change-addr-reg.py

```diff
@@ -0,0 +1,209 @@
+# corresponding on-mesh prefix in the Network Data changes. In this
+# case, the MTD child should schedule an MLE Child Update Request
+# exchange with its parent to re-register the addresses. This ensures
+# that any earlier registration failures due to incorrect context ID
+# compression are resolved.
+#
+
+test_name = __file__[:-3] if __file__.endswith('.py') else __file__
+print('-' * 120)
+print('Starting \'{}\''.format(test_name))
+
+# -----------------------------------------------------------------------------------------------------------------------
+# Creating `cli.Nodes` instances
+
+speedup = 25
+cli.Node.set_time_speedup_factor(speedup)
+
+leader = cli.Node()
+sed = cli.Node()
+
+# -----------------------------------------------------------------------------------------------------------------------
+# Form topology
+
+leader.form('cid-chng')
+
+verify(leader.get_state() == 'leader')
+
+# -----------------------------------------------------------------------------------------------------------------------
+# Test Implementation
+
+# Set the "context ID reuse delay" to a short interval of 3 seconds.
+
+leader.set_context_reuse_delay(3)
+verify(int(leader.get_context_reuse_delay()) == 3)
+
+# Add two on-link prefixes on `leader`.
+
+leader.add_prefix('fd00:1::/64', 'poas')
+leader.add_prefix('fd00:2::/64', 'poas')
+leader.register_netdata()
+
+# Check that context ID 2 is assigned to prefix `fd00:2::/64`.
+
+time.sleep(0.5 / speedup)
+
+netdata = leader.get_netdata()
+contexts = netdata['contexts']
+verify(len(contexts) == 2)
+verify(any([context.startswith('fd00:1:0:0::/64 1 c') for context in contexts]))
+verify(any([context.startswith('fd00:2:0:0::/64 2 c') for context in contexts]))
+
+# Remove the first prefix.
+
+leader.remove_prefix('fd00:1::/64')
+leader.register_netdata()
+
+# Wait longer than Context ID reuse delay to ensure that context ID
+# associated with the removed prefix is aged and removed.
+
+# Wait longer than the Context ID reuse delay to ensure the removed
+# prefix's associated context ID is aged out and removed.
+
+time.sleep(3.5 / speedup)
+
+netdata = leader.get_netdata()
+contexts = netdata['contexts']
+verify(len(contexts) == 1)
+verify(any([context.startswith('fd00:2:0:0::/64 2 c') for context in contexts]))
+
+# Have `sed` attach as a child of `leader`.
+
+sed.set_child_timeout(10)
+sed.set_pollperiod(1000)
+
+sed.join(leader, cli.JOIN_TYPE_SLEEPY_END_DEVICE)
+verify(sed.get_state() == 'child')
+
+verify(int(sed.get_child_timeout()) == 10)
+
+# Check the Network Data on `sed`.
+
+netdata = sed.get_netdata()
+contexts = netdata['contexts']
+verify(len(contexts) == 1)
+verify(any([context.startswith('fd00:2:0:0::/64 2 c') for context in contexts]))
+
+# Find the `sed` address associated with on-mesh prefix `fd00:2::`.
+
+sed_ip_addrs = sed.get_ip_addrs()
+
+for ip_addr in sed_ip_addrs:
+    if ip_addr.startswith('fd00:2:0:0:'):
+        sed_addr = ip_addr
+        break
+else:
+    verify(False)
```

Comment:
nit: indentation?
```suggestion
    else:
        verify(False)
```

Reply:
This is correct and intentional. I am using the python feature where we can have an `else` associated with a `for` loop. https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops

This is used to check if the `for` loop breaks (indicating that we did find a match) and if `for` loop finishes without `break` then `else` will be executed.